Index: html/user_data/plugins/plugin_menu.tpl
===================================================================
--- html/user_data/plugins/plugin_menu.tpl	(リビジョン 0)
+++ html/user_data/plugins/plugin_menu.tpl	(リビジョン 0)
@@ -0,0 +1,35 @@
+<ul id="navi-plugin-menu" class="level1">
+</ul>
+<script type="text/javascript">
+//<![CDATA[
+var host = (("https:" == document.location.protocol) ? "<!--{$smarty.const.SSL_URL}-->" : "<!--{$smarty.const.SITE_URL}-->");
+var pluginURL = host + '<!--{$smarty.const.USER_DIR}--><!--{$smarty.const.PLUGIN_DIR}-->';
+$(function(){
+        $.ajax({
+    url: pluginURL + 'plugins.xml',
+    type: 'GET',
+    dataType: 'xml',
+    timeout: 2000,
+    error: function(){
+        alert("xmlファイルの読み込みに失敗しました");
+    },
+    success: function(xml){
+        $(xml).find("plugin").each(function(){
+            var item_text = $(this).find("name").text();
+            var item_path = $(this).find("path").text();
+
+            $("<li id='navi-plugin-index'></li>")
+               .html("<a href='javascript:;'><span>" + item_text + "</span></a>")
+               .appendTo('ul#navi-plugin-menu')
+               .click(function() {
+                   win03(pluginURL + item_path 
+                          + '/index.php', 'plugins', '800','600');
+                   return false;
+               });
+        });
+        $("li.plugin_menu").html("");
+    }
+    });
+});
+//]]>
+</script>
Index: html/user_data/plugins/plugins.xml
===================================================================
--- html/user_data/plugins/plugins.xml	(リビジョン 0)
+++ html/user_data/plugins/plugins.xml	(リビジョン 0)
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<plugins>
+  <plugin>
+    <name>Google Analytics</name>
+    <path>google_analytics</path>
+  </plugin>
+</plugins>
Index: html/user_data/plugins/google_analytics/sql/delete.sql
===================================================================
--- html/user_data/plugins/google_analytics/sql/delete.sql	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/sql/delete.sql	(リビジョン 0)
@@ -0,0 +1,2 @@
+DELETE FROM dtb_bloc WHERE bloc_name = 'Google Analytics';
+DELETE FROM dtb_blocposition WHERE bloc_id = (SELECT bloc_id FROM dtb_bloc WHERE bloc_name = 'Google Analytics');
Index: html/user_data/plugins/google_analytics/sql/insert.sql
===================================================================
--- html/user_data/plugins/google_analytics/sql/insert.sql	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/sql/insert.sql	(リビジョン 0)
@@ -0,0 +1 @@
+INSERT INTO dtb_bloc (bloc_name, tpl_path, filename, php_path, del_flg) VALUES ('Google Analytics', 'user_data/plugins/google_analytics/tpl/ga.tpl', 'google_analytics', 'user_data/plugins/google_analytics/ga.php', 1);
\ No newline at end of file
Index: html/user_data/plugins/google_analytics/tpl/ga.tpl
===================================================================
--- html/user_data/plugins/google_analytics/tpl/ga.tpl	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/tpl/ga.tpl	(リビジョン 0)
@@ -0,0 +1,9 @@
+<script type="text/javascript">
+var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
+document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
+</script>
+<script type="text/javascript">
+try {
+var pageTracker = _gat._getTracker("UA-<!--{$smarty.const.GA_UA}-->");
+pageTracker._trackPageview();
+} catch(err) {}</script>
Index: html/user_data/plugins/google_analytics/tpl/index.tpl
===================================================================
--- html/user_data/plugins/google_analytics/tpl/index.tpl	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/tpl/index.tpl	(リビジョン 0)
@@ -0,0 +1,52 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="ja" lang="ja">
+
+<head>
+<meta http-equiv="content-type" content="application/xhtml+xml; charset=<!--{$smarty.const.CHAR_CODE}-->" />
+<meta http-equiv="content-script-type" content="text/javascript" />
+<meta http-equiv="content-style-type" content="text/css" />
+<link rel="stylesheet" href="<!--{$TPL_DIR}-->css/admin_contents.css" type="text/css" media="all" />
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/css.js"></script>
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/navi.js"></script>
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/win_op.js"></script>
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/site.js"></script>
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/jquery.js"></script>
+<script type="text/javascript" src="<!--{$TPL_DIR}-->js/admin.js"></script>
+<title><!--{$tpl_subtitle}--></title>
+<script type="text/javascript">
+
+</script>
+<style type="text/css">
+@charset "UTF-8";
+</style>
+</head>
+
+<body onload="<!--{$tpl_onload}-->" class="authority_0" style="min_width: 400px">
+<noscript>
+  <p>JavaScript を有効にしてご利用下さい.</p>
+</noscript>
+
+<div id="container">
+<a name="top"></a>
+<!--▼CONTENTS-->
+<h1>Google Analytics Plugin</h1>
+<div id="contents" class="clear-block">
+  <form method="post" action="index.php">
+    <div id="system" class="contents-main">
+      <table class="list">
+	<tr>
+	  <th>ウェブ プロパティ ID</th>
+	  <td>UA-<input type="text" name="ga_ua" value="<!--{$smarty.const.GA_UA}-->" /></td>
+	</tr>
+      </table>
+      <div class="btn addnew">
+	<input type="hidden" name="mode" value="register" />
+	<button type="submit"><span>この内容で登録する</span></button>
+      </div>
+    </div>
+  </form>
+</div>
+<!--▲CONTENTS-->
+</div>
+</body>
+</html>
Index: html/user_data/plugins/google_analytics/require.php
===================================================================
--- html/user_data/plugins/google_analytics/require.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/require.php	(リビジョン 0)
@@ -0,0 +1,6 @@
+<?php
+/*
+ * EC-CUBE 本体で, このプラグインの関数を使用したい場合は, ここで require する
+ */
+require_once(PLUGIN_PATH . "google_analytics/classes/pages/ga_config.php");
+?>
\ No newline at end of file
Index: html/user_data/plugins/google_analytics/classes/pages/ga_config.php
===================================================================
--- html/user_data/plugins/google_analytics/classes/pages/ga_config.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/classes/pages/ga_config.php	(リビジョン 0)
@@ -0,0 +1,3 @@
+<?php
+define('GA_UA', '');
+?>
Index: html/user_data/plugins/google_analytics/classes/pages/LC_Page_FrontParts_Bloc_GoogleAnalytics.php
===================================================================
--- html/user_data/plugins/google_analytics/classes/pages/LC_Page_FrontParts_Bloc_GoogleAnalytics.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/classes/pages/LC_Page_FrontParts_Bloc_GoogleAnalytics.php	(リビジョン 0)
@@ -0,0 +1,70 @@
+<?php
+/*
+ * This file is part of EC-CUBE
+ *
+ * Copyright(c) 2000-2009 LOCKON CO.,LTD. All Rights Reserved.
+ *
+ * http://www.lockon.co.jp/
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+// {{{ requires
+require_once CLASS_PATH . "pages/LC_Page.php";
+
+/**
+ * Google Analytics プラグインを制御するクラス.
+ *
+ * @package Page
+ * @author Kentaro Ohkouchi
+ * @version $Id$
+ */
+class LC_Page_FrontParts_Bloc_GoogleAnalytics extends LC_Page {
+
+    // }}}
+    // {{{ functions
+
+    /**
+     * Page を初期化する.
+     *
+     * @return void
+     */
+    function init() {
+        parent::init();
+        $this->tpl_mainpage = USER_PATH . 'plugins/google_analytics/tpl/ga.tpl';
+        $this->tpl_subtitle = "Google Analytics Plugin";
+    }
+
+    /**
+     * Page のプロセス.
+     *
+     * @return void
+     */
+    function process() {
+        $objView = new SC_SiteView();
+        $objView->assignobj($this);
+        $objView->display($this->tpl_mainpage);
+    }
+
+    /**
+     * デストラクタ.
+     *
+     * @return void
+     */
+    function destroy() {
+        parent::destroy();
+    }
+}
+?>
\ No newline at end of file
Index: html/user_data/plugins/google_analytics/classes/pages/LC_Page_Admin_GoogleAnalytics.php
===================================================================
--- html/user_data/plugins/google_analytics/classes/pages/LC_Page_Admin_GoogleAnalytics.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/classes/pages/LC_Page_Admin_GoogleAnalytics.php	(リビジョン 0)
@@ -0,0 +1,141 @@
+<?php
+/*
+ * This file is part of EC-CUBE
+ *
+ * Copyright(c) 2000-2009 LOCKON CO.,LTD. All Rights Reserved.
+ *
+ * http://www.lockon.co.jp/
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+// {{{ requires
+require_once CLASS_PATH . "pages/LC_Page.php";
+
+/**
+ * Google Analytics プラグインの管理画面を制御するクラス.
+ *
+ * @package Page
+ * @author Kentaro Ohkouchi
+ * @version $Id$
+ */
+class LC_Page_Admin_GoogleAnalytics extends LC_Page {
+
+    // }}}
+    // {{{ functions
+
+    /**
+     * Page を初期化する.
+     *
+     * @return void
+     */
+    function init() {
+        parent::init();
+        $this->tpl_mainpage = PLUGIN_PATH . 'google_analytics/tpl/index.tpl';
+        $this->tpl_subtitle = "Google Analytics Plugin";
+
+        if (empty($_POST["mode"])) {
+            $_POST["mode"] = "";
+        }
+
+        if (empty($_GET["mode"])) {
+            $_GET["mode"] = "";
+        }
+    }
+
+    /**
+     * Page のプロセス.
+     *
+     * POST パラメータ "mode" が register の場合は登録処理を行う.
+     * 登録処理の後, 自ページをリロードし, GET パラメータ "mode" を付与する.
+     * 登録に成功し, GET パラメータ "mode" の値が success の場合は
+     * 「登録に成功しました」というメッセージをポップアップで表示する.
+     * 登録に失敗し, GET パラメータ "mode" の値が failure の場合は
+     * 「登録に失敗しました」というメッセージをポップアップで表示する.
+     *
+     * TODO Transaction Token を使用する
+     *
+     * @return void
+     */
+    function process() {
+        // 認証可否の判定
+        SC_Utils_Ex::sfIsSuccess(new SC_Session());
+
+        switch ($_POST["mode"]) {
+        case "register":
+            if ($this->register($_POST['ga_ua'])) {
+                $this->reload(array("mode" => "success"));
+                exit;
+            } else {
+                $this->reload(array("mode" => "failure"));
+                exit;
+            }
+            break;
+
+          default:
+        }
+
+        switch ($_GET["mode"]) {
+        case "success":
+            $this->tpl_onload .= "window.alert('登録に成功しました。');";
+            break;
+
+        case "failure":
+            $this->tpl_onload .= "window.alert('登録に失敗しました。');";
+            break;
+
+          default:
+        }
+
+        $objView = new SC_AdminView();
+        $objView->assignobj($this);
+        $objView->display($this->tpl_mainpage);
+    }
+
+    /**
+     * UA の登録を行う.
+     *
+     * classes/pages/ga_config.php を読み込み, ウェブプロパティID の文字列
+     * を定数として書き出す.
+     *
+     * @param string ウェブプロパティID の文字列
+     * @return boolean 登録に成功した場合 true; 失敗した場合 false;
+     */
+    function register($ua) {
+        $data = "<?php\ndefine('GA_UA', '" 
+            . htmlspecialchars($ua, ENT_QUOTES) . "');\n?>\n";
+
+        $configFile = PLUGIN_PATH . "google_analytics/classes/pages/ga_config.php";
+        $handle = fopen($configFile, "w");
+        if (!$handle) {
+            return false;
+        }
+        // ファイルの内容を書き出す.
+        if (fwrite($handle, $data) === false) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * デストラクタ.
+     *
+     * @return void
+     */
+    function destroy() {
+        parent::destroy();
+    }
+}
+?>
\ No newline at end of file
Index: html/user_data/plugins/google_analytics/ga.php
===================================================================
--- html/user_data/plugins/google_analytics/ga.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/ga.php	(リビジョン 0)
@@ -0,0 +1,12 @@
+<?php
+// {{{ requires
+require_once(PLUGIN_PATH . "google_analytics/classes/pages/LC_Page_FrontParts_Bloc_GoogleAnalytics.php");
+
+// }}}
+// {{{ generate page
+
+$objPage = new LC_Page_FrontParts_Bloc_GoogleAnalytics();
+register_shutdown_function(array($objPage, "destroy"));
+$objPage->init();
+$objPage->process();
+?>
\ No newline at end of file
Index: html/user_data/plugins/google_analytics/index.php
===================================================================
--- html/user_data/plugins/google_analytics/index.php	(リビジョン 0)
+++ html/user_data/plugins/google_analytics/index.php	(リビジョン 0)
@@ -0,0 +1,34 @@
+<?php
+/*
+ * This file is part of EC-CUBE
+ *
+ * Copyright(c) 2000-2007 LOCKON CO.,LTD. All Rights Reserved.
+ *
+ * http://www.lockon.co.jp/
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+// {{{ requires
+require_once("../../../admin/require.php");
+require_once(USER_PATH . 'plugins/google_analytics/classes/pages/LC_Page_Admin_GoogleAnalytics.php');
+
+// }}}
+// {{{ generate page
+
+$objPage = new LC_Page_Admin_GoogleAnalytics();
+register_shutdown_function(array($objPage, 'destroy'));
+$objPage->init();
+$objPage->process();
+?>
\ No newline at end of file
Index: html/install/sql/insert_data.sql
===================================================================
--- html/install/sql/insert_data.sql	(リビジョン 18229)
+++ html/install/sql/insert_data.sql	(作業コピー)
@@ -1114,5 +1114,8 @@
 INSERT INTO mtb_constants VALUES ('OSTORE_E_C_BATCH_ERR', '"2010"', 522, 'オーナーズストア通信エラーコード');
 INSERT INTO mtb_constants VALUES ('OPTION_FAVOFITE_PRODUCT', '1', 523, 'お気に入り商品登録(有効:1 無効:0)');
 INSERT INTO mtb_constants VALUES ('IMAGE_RENAME', 'true', 525, '画像リネーム設定（商品画像のみ）(true:リネームする、false:リネームしない)');
+INSERT INTO mtb_constants VALUES ('PLUGIN_DIR', '"plugins/"', 600, 'プラグインディレクトリ');
+INSERT INTO mtb_constants VALUES ('PLUGIN_PATH', 'USER_PATH . PLUGIN_DIR', 601, 'プラグイン保存先');
+INSERT INTO mtb_constants VALUES ('PLUGIN_URL', 'USER_URL . PLUGIN_DIR', 602, 'プラグイン URL');
 
 INSERT INTO dtb_module (module_id,module_code,module_name,update_date,create_date) VALUES (0,0,'patch',now(),now());
Index: data/Smarty/templates/default/admin/main_frame.tpl
===================================================================
--- data/Smarty/templates/default/admin/main_frame.tpl	(リビジョン 18229)
+++ data/Smarty/templates/default/admin/main_frame.tpl	(作業コピー)
@@ -113,6 +113,10 @@
         <a><span>OWNERS STORE</span></a>
         <!--{include file="`$smarty.const.TEMPLATE_ADMIN_DIR`ownersstore/subnavi.tpl"}-->
     </li>
+    <li id="navi-plugin" class="<!--{if $tpl_mainno eq "plugin"}-->on<!--{/if}-->">
+      <a><span>プラグイン設定</span></a>
+      <!--{include file="`$smarty.const.PLUGIN_PATH`plugin_menu.tpl"}-->
+    </li>
 </ul>
 <div style="clear: both;"></div>
 <!--{* ▲NAVI *}-->
Index: data/require_plugin.php
===================================================================
--- data/require_plugin.php	(リビジョン 18229)
+++ data/require_plugin.php	(作業コピー)
@@ -1,5 +1,39 @@
 <?php
-  /*
-   * プラグインを require するためのファイル
-   */
+/*
+ * This file is part of EC-CUBE
+ *
+ * Copyright(c) 2000-2009 LOCKON CO.,LTD. All Rights Reserved.
+ *
+ * http://www.lockon.co.jp/
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+// load plugins
+/* -----------------------------------------------------------------------------
+ * TODO PHP4 でも使えるように, XML パーサーをファクトリークラスで実装する
+ * ----------------------------------------------------------------------------*/
+if (version_compare("5", PHP_VERSION, "<")) {
+    $plugins = file_get_contents(USER_PATH . "plugins/plugins.xml");
+    $xml = new SimpleXMLElement($plugins);
+    foreach ($xml->plugin as $plugin) {
+    
+        $requireFile = USER_PATH . "plugins/" . $plugin->path . "/require.php";
+        if (file_exists($requireFile)) {
+            require_once($requireFile);
+        }
+    }
+}
 ?>
Index: data/mtb_constants_init.php
===================================================================
--- data/mtb_constants_init.php	(リビジョン 18229)
+++ data/mtb_constants_init.php	(作業コピー)
@@ -670,4 +670,10 @@
 define('OPTION_FAVOFITE_PRODUCT','1');
 /** 画像リネーム設定（商品画像のみ） */
 define('IMAGE_RENAME', true);
+/** プラグインディレクトリ */
+define("PLUGIN_DIR", "plugins/");
+/** プラグイン保存先 */
+define("PLUGIN_PATH", USER_PATH . PLUGIN_DIR);
+/** プラグイン URL */
+define("PLUGIN_URL", USER_URL . PLUGIN_DIR);
 ?>
Index: data/require_base.php
===================================================================
--- data/require_base.php	(リビジョン 18229)
+++ data/require_base.php	(作業コピー)
@@ -78,7 +78,6 @@
 require_once(CLASS_EX_PATH . "helper_extends/SC_Helper_Session_Ex.php");
 require_once(CLASS_EX_PATH . "helper_extends/SC_Helper_Mail_Ex.php");
 require_once(CLASS_EX_PATH . "helper_extends/SC_Helper_Mobile_Ex.php");
-include_once(DATA_PATH . "require_plugin.php");
 
 // セッションハンドラ開始
 $objSession = new SC_Helper_Session_Ex();
@@ -91,6 +90,9 @@
 $sessionFactory = SC_SessionFactory::getInstance();
 $sessionFactory->initSession();
 
+// プラグインを読み込む
+include_once(DATA_PATH . "require_plugin.php");
+
 // 絵文字変換 (除去) フィルターを組み込む。
 ob_start(array('SC_MobileEmoji', 'handler'));
 ?>
Index: data/class/pages/LC_Page.php
===================================================================
--- data/class/pages/LC_Page.php	(リビジョン 18229)
+++ data/class/pages/LC_Page.php	(作業コピー)
@@ -82,6 +82,9 @@
      */
     function init() {
         $this->tpl_authority = $_SESSION['authority'];
+        // XXX すべてのページで宣言するべき
+        $layout = new SC_Helper_PageLayout_Ex();
+        $layout->sfGetPageLayout($this, false);
     }
 
     /**
Index: data/class/helper/SC_Helper_PageLayout.php
===================================================================
--- data/class/helper/SC_Helper_PageLayout.php	(リビジョン 18229)
+++ data/class/helper/SC_Helper_PageLayout.php	(作業コピー)
@@ -61,9 +61,11 @@
             $objPage->tpl_mainpage = USER_PATH . "templates/preview/"
                 . TEMPLATE_NAME . "/" . $arrPageData[0]['filename'] . ".tpl";
         }
-                
-        foreach($arrPageData[0] as $key => $val) {
-            $debug_message.= "arrPageData[$key]：" . $val . "\n";
+
+        if (!empty($arrPageData[0])) {
+            foreach($arrPageData[0] as $key => $val) {
+                $debug_message.= "arrPageData[$key]：" . $val . "\n";
+            }
         }
         
         $debug_message.= "TEMPLATE_NAME：".TEMPLATE_NAME . "\n";
