=== modified file 'data/module/MDB2/Driver/pgsql.php'
--- data/module/MDB2/Driver/pgsql.php	2010-07-06 08:39:45 +0000
+++ data/module/MDB2/Driver/pgsql.php	2010-11-18 05:39:27 +0000
@@ -929,7 +929,9 @@
                     } elseif (is_array($types) && array_key_exists($parameter, $types)) {
                         $pgtypes[] = $this->datatype->mapPrepareDatatype($types[$parameter]);
                     } else {
-                        $pgtypes[] = 'text';
+                        if (version_compare(PHP_VERSION, '5.0.0', '>')) {
+                            $pgtypes[] = 'text';
+                        }
                     }
                 }
                 if (($key_parameter = array_search($name, $positions))) {

