Index: Mobile.php

--- data/module/Net/UserAgent/Mobile.php ベース (BASE)
+++ data/module/Net/UserAgent/Mobile.php ローカルでの変更 (LOCAL がベース)
@@ -424,7 +424,8 @@
             'blackberry9800', // Torch
             'webOS',          // Palm Pre Experimental
             'incognito',      // Other iPhone browser
-            'webmate'         // Other iPhone browser
+            'webmate',         // Other iPhone browser
+            'Windows Phone OS' // Windows Phone
         );
 
         $pattern = implode("|", $useragents);
