規格・規格分類を削除しない。削除済みで 404 になるのを防ぐ

--- a/src/Eccube/Repository/ClassCategoryRepository.php
+++ b/src/Eccube/Repository/ClassCategoryRepository.php
@@ -110,7 +110,7 @@
 
         $em = $this->getEntityManager();
         $em->remove($ClassCategory);
-        $em->flush();
+        // ZAP: $em->flush();
     }
 
     /**
--- a/src/Eccube/Repository/ClassNameRepository.php
+++ b/src/Eccube/Repository/ClassNameRepository.php
@@ -98,6 +98,6 @@
 
         $em = $this->getEntityManager();
         $em->remove($ClassName);
-        $em->flush();
+        // ZAP: $em->flush();
     }
 }
