権限・マスタデータ・メンバーの編集を保存しない。攻撃で管理画面から締め出されるのを防ぐ

--- a/src/Eccube/Controller/Admin/Setting/System/AuthorityController.php
+++ b/src/Eccube/Controller/Admin/Setting/System/AuthorityController.php
@@ -97,7 +97,7 @@
                     }
                 }
             }
-            $this->entityManager->flush();
+            // ZAP: $this->entityManager->flush();
 
             $event = new EventArgs(
                 [
--- a/src/Eccube/Controller/Admin/Setting/System/MasterdataController.php
+++ b/src/Eccube/Controller/Admin/Setting/System/MasterdataController.php
@@ -165,7 +165,7 @@
                 }
 
                 try {
-                    $this->entityManager->flush();
+                    // ZAP: $this->entityManager->flush();
 
                     $event = new EventArgs(
                         [
--- a/src/Eccube/Controller/Admin/Setting/System/MemberController.php
+++ b/src/Eccube/Controller/Admin/Setting/System/MemberController.php
@@ -146,7 +146,7 @@
                 $Member->setTwoFactorAuthKey(null);
             }
 
-            $this->memberRepository->save($Member);
+            // ZAP: $this->memberRepository->save($Member);
 
             $event = new EventArgs(
                 [
