会員登録を保存しない。登録済みのメールと重複して、完了への攻撃が入力画面に戻るのを防ぐ

--- a/src/Eccube/Controller/EntryController.php
+++ b/src/Eccube/Controller/EntryController.php
@@ -111,7 +111,7 @@
                     $Customer->setPassword($password);
 
                     $this->entityManager->persist($Customer);
-                    $this->entityManager->flush();
+                    // ZAP: $this->entityManager->flush();
 
                     log_info('会員登録完了');
 
